app/web_framework/cluster_base_handler.py

```diff
@@ -86,6 +89,10 @@ def prepare(self):
         """
         self._check_expected_session_id()
 
+        accept_header = self.request.headers.get('Accept')
+        uri = self.request.uri
+        self.api_version = APIVersionHandler.resolve_version(accept_header, uri)
```

Comment:
> Should this be in the superclass (ClusterBaseHandler)?

Yeah maybe, I wasn't too sure where exactly this should live. I was looking at the [tornado request handling docs](http://www.tornadoweb.org/en/stable/guide/structure.html#overriding-requesthandler-methods) and trying to figure out _when_ is the best part to parse the headers. It looks like we can (and should) add it to the `initialize` function of the `ClusterBaseHandler`, is this right?